File: src/contests/cg/puzzle/distributing-candy.ts
// 🎮 CodinGame Puzzle - distributing-candy
// https://www.codingame.com/training/easy/distributing-candy

const [n, m] = readline().split(" ").map(Number)
const c = readline().split(" ").map(Number)
c.sort((a, b) => a - b)
let best = Infinity
for (let i = 0; i + m <= n; i++) {
  best = Math.min(best, c[i + m - 1] - c[i])
}
console.log(best)

File: src/contests/cg/puzzle/clojure/distributing-candy.clj
;; 🎮 CodinGame Puzzle - distributing-candy
;; https://www.codingame.com/training/easy/distributing-candy

(ns Solution (:gen-class))

(defn -main [& _]
  (let [parts (.split (.trim (read-line)) "\\s+")
        n (Integer/parseInt (aget parts 0))
        m (Integer/parseInt (aget parts 1))
        candies (vec (sort (map #(Integer/parseInt %) (.split (.trim (read-line)) "\\s+"))))]
    (loop [i 0 best Long/MAX_VALUE]
      (if (< (+ i m -1) n)
        (recur (inc i) (min best (- (nth candies (+ i m -1)) (nth candies i))))
        (println best))))
  (flush))
